app/root.js

```diff
@@ -21,6 +21,65 @@ function mapStateToProps (state) {
   }
 }
 
+function isVaultValid (vault) {
+  return typeof vault === 'object'
+    && ['data', 'iv', 'salt'].every(e => typeof vault[e] === 'string')
+}
+
+// Deduplicates array with rudimentary non-recursive shallow comparison of keys
+function dedupe (arr) {
+  const result = []
+  arr.forEach(x => {
+    if (!result.find(y => Object.keys(x).length === Object.keys(y).length && Object.entries(x).every(([k,ex]) => y[k] === ex ))) {
+      result.push(x)
+    }
+  })
+  return result
+}
+
+function extractVaultFromFile (data) {
+  let vaultBody
+  try {
+    // attempt 1: raw json
+    return JSON.parse(data)
+  } catch (err) {
+    // Not valid JSON: continue
+  }
+  {
+    // attempt 2: chromium 000003.log file on linux
+    const matches = data.match(/"KeyringController":{"vault":"{[^{}]*}"/)
+    if (matches && matches.length) {
+      vaultBody=matches[0].substring(29)
+      return JSON.parse(
+        JSON.parse(
+          vaultBody
+        )
+      )
+    }
+  }
+  // attempt 3: chromium 000005.ldb on windows
```

Comment:
(optional) would it be a lot of work to drop example files from dummy vaults as fixtures for a test so that it's easier to explore what each attempt is doing?

Reply:
It's a good idea, as noted by @mcmire as well. Agreed this is desired - will make sure to add that in a follow-up.